src/vs/workbench/contrib/terminal/browser/terminalTabsList.ts

```diff
@@ -607,54 +610,68 @@ class TerminalTabsDragAndDrop implements IListDragAndDrop<ITerminalInstance> {
 		};
 	}
 
-	drop(data: IDragAndDropData, targetInstance: ITerminalInstance | undefined, targetIndex: number | undefined, originalEvent: DragEvent): void {
+	async drop(data: IDragAndDropData, targetInstance: ITerminalInstance | undefined, targetIndex: number | undefined, originalEvent: DragEvent): Promise<void> {
 		this._autoFocusDisposable.dispose();
 		this._autoFocusInstance = undefined;
 
 		let sourceInstances: ITerminalInstance[] | undefined;
-		const terminalResources = originalEvent.dataTransfer?.getData(DataTransfers.TERMINALS);
-		if (terminalResources) {
+		const terminalResources = originalEvent.dataTransfer?.getData(DataTransfers.RESOURCES);
+		const terminals = originalEvent.dataTransfer?.getData(DataTransfers.TERMINALS);
+		if (terminals && terminalResources) {
```

Comment:
`!terminals || !terminalResources`?

Reply:
will a file dropped on a terminal tab meet those criteria?

Reply:
Was just thinking we could invert it as a huge indentation change happened below

Reply:
figured out the cause of the indentation - an inappropriately placed end bracket. fixed - good catch.